diff --git a/src/AppBundle/Factory/DinosaurFactory.php b/src/AppBundle/Factory/DinosaurFactory.php
index f36eb18..bb9bef7 100644
--- a/src/AppBundle/Factory/DinosaurFactory.php
+++ b/src/AppBundle/Factory/DinosaurFactory.php
@@ -26,6 +26,10 @@ class DinosaurFactory
             $length = random_int(Dinosaur::LARGE, 100);
         }
 
+        if (stripos($specification, 'carnivorous') !== false) {
+            $isCarnivorous = true;
+        }
+
         $dinosaur = new Dinosaur($codeName, $isCarnivorous);
 
         $dinosaur->setLength($length);
