diff --git a/src/AppBundle/Factory/DinosaurFactory.php b/src/AppBundle/Factory/DinosaurFactory.php
index e666f88..ab8599d 100644
--- a/src/AppBundle/Factory/DinosaurFactory.php
+++ b/src/AppBundle/Factory/DinosaurFactory.php
@@ -22,5 +22,13 @@ class DinosaurFactory
     
     public function growFromSpecification(string $specification): Dinosaur
     {
+        // defaults
+        $codeName = 'InG-' . random_int(1, 99999);
+        $length = random_int(1, Dinosaur::LARGE - 1);
+        $isCarnivorous = false;
+
+        $dinosaur = $this->createDinosaur($codeName, $isCarnivorous, $length);
+
+        return $dinosaur;
     }
 }
