diff --git a/src/AppBundle/Factory/DinosaurFactory.php b/src/AppBundle/Factory/DinosaurFactory.php
index ab8599d..613d053 100644
--- a/src/AppBundle/Factory/DinosaurFactory.php
+++ b/src/AppBundle/Factory/DinosaurFactory.php
@@ -27,6 +27,10 @@ class DinosaurFactory
         $length = random_int(1, Dinosaur::LARGE - 1);
         $isCarnivorous = false;
 
+        if (stripos($specification, 'large') !== false) {
+            $length = random_int(Dinosaur::LARGE, 100);
+        }
+
         $dinosaur = $this->createDinosaur($codeName, $isCarnivorous, $length);
 
         return $dinosaur;
