diff --git a/src/AppBundle/Factory/DinosaurFactory.php b/src/AppBundle/Factory/DinosaurFactory.php
index e85d6f7..02fc411 100644
--- a/src/AppBundle/Factory/DinosaurFactory.php
+++ b/src/AppBundle/Factory/DinosaurFactory.php
@@ -27,8 +27,12 @@ class DinosaurFactory
         $length = random_int(1, Dinosaur::LARGE - 1);
         $isCarnivorous = false;
 
+        if (stripos($specification, 'huge') !== false) {
+            $length = random_int(Dinosaur::HUGE, 100);
+        }
+
         if (stripos($specification, 'large') !== false) {
-            $length = random_int(Dinosaur::LARGE, 100);
+            $length = random_int(Dinosaur::LARGE, Dinosaur::HUGE - 1);
         }
 
         if (stripos($specification, 'carnivorous') !== false) {
