diff --git a/tests/AppBundle/Factory/DinosaurFactoryTest.php b/tests/AppBundle/Factory/DinosaurFactoryTest.php
index 3d14ea4..e9f524b 100644
--- a/tests/AppBundle/Factory/DinosaurFactoryTest.php
+++ b/tests/AppBundle/Factory/DinosaurFactoryTest.php
@@ -65,7 +65,7 @@ class DinosaurFactoryTest extends TestCase
         return [
             // specification, is large, is carnivorous
             ['large carnivorous dinosaur', true, true],
-            ['give me all the cookies!!!', false, false],
+            'default response' => ['give me all the cookies!!!', false, false],
             ['large herbivore', true, false],
         ];
     }
