diff --git a/src/AppBundle/Entity/Enclosure.php b/src/AppBundle/Entity/Enclosure.php
new file mode 100644
index 0000000..0c6f7de
--- /dev/null
+++ b/src/AppBundle/Entity/Enclosure.php
@@ -0,0 +1,25 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+class Enclosure
+{
+    /**
+     * @var Collection
+     */
+    private $dinosaurs;
+
+    public function __construct()
+    {
+        $this->dinosaurs = new ArrayCollection();
+    }
+
+    public function getDinosaurs(): Collection
+    {
+        return $this->dinosaurs;
+    }
+}
diff --git a/tests/AppBundle/Entity/EnclosureTest.php b/tests/AppBundle/Entity/EnclosureTest.php
new file mode 100644
index 0000000..0cba364
--- /dev/null
+++ b/tests/AppBundle/Entity/EnclosureTest.php
@@ -0,0 +1,16 @@
+<?php
+
+namespace Tests\AppBundle\Entity;
+
+use AppBundle\Entity\Enclosure;
+use PHPUnit\Framework\TestCase;
+
+class EnclosureTest extends TestCase
+{
+    public function testItHasNoDinosaursByDefault()
+    {
+        $enclosure = new Enclosure();
+
+        $this->assertCount(0, $enclosure->getDinosaurs());
+    }
+}
