diff --git a/tests/AppBundle/Entity/EnclosureTest.php b/tests/AppBundle/Entity/EnclosureTest.php
index 6a10526..eac6c7e 100644
--- a/tests/AppBundle/Entity/EnclosureTest.php
+++ b/tests/AppBundle/Entity/EnclosureTest.php
@@ -36,4 +36,15 @@ class EnclosureTest extends TestCase
 
         $enclosure->addDinosaur(new Dinosaur('Velociraptor', true));
     }
+
+    /**
+     * @expectedException \AppBundle\Exception\NotABuffetException
+     */
+    public function testItDoesNotAllowToAddNonCarnivorousDinosaursToCarnivorousEnclosure()
+    {
+        $enclosure = new Enclosure();
+
+        $enclosure->addDinosaur(new Dinosaur('Velociraptor', true));
+        $enclosure->addDinosaur(new Dinosaur());
+    }
 }
