diff --git a/src/AppBundle/Entity/Enclosure.php b/src/AppBundle/Entity/Enclosure.php
index c5026cf..98f2ce5 100644
--- a/src/AppBundle/Entity/Enclosure.php
+++ b/src/AppBundle/Entity/Enclosure.php
@@ -58,6 +58,10 @@ class Enclosure
 
     public function addSecurity($security)
     {
+        if (!$security instanceof Security) {
+            throw new \InvalidArgumentException('This does not look safe');
+        }
+
         $this->securities[] = $security;
     }
 }
