diff --git a/src/AppBundle/Entity/Enclosure.php b/src/AppBundle/Entity/Enclosure.php
index 5cca59d..62548fc 100644
--- a/src/AppBundle/Entity/Enclosure.php
+++ b/src/AppBundle/Entity/Enclosure.php
@@ -24,10 +24,17 @@ class Enclosure
      */
     private $isCarnivorous;
 
+    /**
+     * @var iterable
+     * @ORM\OneToMany(targetEntity="AppBundle\Entity\Security", mappedBy="enclosure", cascade={"persist"})
+     */
+    private $securities;
+
     public function __construct(bool $isCarnivorous = false)
     {
         $this->dinosaurs = new ArrayCollection();
         $this->isCarnivorous = $isCarnivorous;
+        $this->securities = new ArrayCollection();
     }
 
     public function getDinosaurs(): iterable
@@ -48,4 +55,9 @@ class Enclosure
     {
         return $this->isCarnivorous;
     }
+
+    public function addSecurity($security): void
+    {
+        $this->securities[] = $security;
+    }
 }
diff --git a/src/AppBundle/Entity/Security.php b/src/AppBundle/Entity/Security.php
new file mode 100644
index 0000000..d35a5f4
--- /dev/null
+++ b/src/AppBundle/Entity/Security.php
@@ -0,0 +1,46 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="securities")
+ */
+class Security
+{
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $name;
+
+    /**
+     * @ORM\Column(type="boolean")
+     */
+    private $isActive;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Enclosure")
+     */
+    private $enclosure;
+
+    public function __construct(string $name, bool $isActive, Enclosure $enclosure)
+    {
+        $this->name = $name;
+        $this->isActive = $isActive;
+        $this->enclosure = $enclosure;
+    }
+
+    public function getIsActive(): bool
+    {
+        return $this->isActive;
+    }
+}
diff --git a/tests/AppBundle/Entity/EnclosureTest.php b/tests/AppBundle/Entity/EnclosureTest.php
index 1f7f647..e7fd14d 100644
--- a/tests/AppBundle/Entity/EnclosureTest.php
+++ b/tests/AppBundle/Entity/EnclosureTest.php
@@ -47,4 +47,14 @@ class EnclosureTest extends TestCase
         $enclosure->addDinosaur(new Dinosaur('Velociraptor', true));
         $enclosure->addDinosaur(new Dinosaur());
     }
+
+    public function testItDoesNotAllowUnsafeSecurities()
+    {
+        $enclosure = new Enclosure();
+
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('This does not look safe');
+
+        $enclosure->addSecurity(new Dinosaur('Velociraptor', true));
+    }
 }
