diff --git a/tests/AppBundle/Controller/DefaultControllerTest.php b/tests/AppBundle/Controller/DefaultControllerTest.php
index 56aedb7..cad81f1 100644
--- a/tests/AppBundle/Controller/DefaultControllerTest.php
+++ b/tests/AppBundle/Controller/DefaultControllerTest.php
@@ -50,6 +50,7 @@ class DefaultControllerTest extends WebTestCase
         ]);
 
         $client = $this->makeClient();
+	$client->followRedirects();
 
         $crawler = $client->request('GET', '/');
 
