diff --git a/src/AppBundle/Entity/Enclosure.php b/src/AppBundle/Entity/Enclosure.php
index 018f522..8e21630 100644
--- a/src/AppBundle/Entity/Enclosure.php
+++ b/src/AppBundle/Entity/Enclosure.php
@@ -42,6 +42,11 @@ class Enclosure
         }
     }
 
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
     public function getDinosaurs(): Collection
     {
         return $this->dinosaurs;
diff --git a/tests/AppBundle/Controller/DefaultControllerTest.php b/tests/AppBundle/Controller/DefaultControllerTest.php
index e3e789a..f15e8bd 100644
--- a/tests/AppBundle/Controller/DefaultControllerTest.php
+++ b/tests/AppBundle/Controller/DefaultControllerTest.php
@@ -10,15 +10,20 @@ class DefaultControllerTest extends WebTestCase
 {
     public function testEnclosuresAreShownOnHomepage()
     {
-        $this->loadFixtures([
+        $fixtures = $this->loadFixtures([
             LoadBasicParkData::class,
             LoadSecurityData::class,
-        ]);
+        ])->getReferenceRepository();
 
         $client = $this->makeClient();
 
         $crawler = $client->request('GET', '/');
 
         $this->assertStatusCode(200, $client);
+
+        $enclosure = $fixtures->getReference('carnivorous-enclosure');
+        $selector = '#enclosure-' . $enclosure->getId() . ' .button-alarm';
+
+        $this->assertGreaterThan(0, $crawler->filter($selector)->count());
     }
 }
