diff --git a/tests/AppBundle/Controller/DefaultControllerTest.php b/tests/AppBundle/Controller/DefaultControllerTest.php
index 978b65b..56aedb7 100644
--- a/tests/AppBundle/Controller/DefaultControllerTest.php
+++ b/tests/AppBundle/Controller/DefaultControllerTest.php
@@ -41,4 +41,18 @@ class DefaultControllerTest extends WebTestCase
 
         $this->assertGreaterThan(0, $crawler->filter($selector)->count());
     }
+
+    public function testItGrowsADinosaurFromSpecification()
+    {
+        $this->loadFixtures([
+            LoadBasicParkData::class,
+            LoadSecurityData::class,
+        ]);
+
+        $client = $this->makeClient();
+
+        $crawler = $client->request('GET', '/');
+
+        $this->assertStatusCode(200, $client);
+    }
 }
