diff --git a/app/Resources/views/default/index.html.twig b/app/Resources/views/default/index.html.twig
index 3fabf3b..e61ca97 100644
--- a/app/Resources/views/default/index.html.twig
+++ b/app/Resources/views/default/index.html.twig
@@ -13,6 +13,13 @@
                     <td>
                         Contains <strong>{{ enclosure.dinosaurCount }}</strong> dinosaur(s)
                     </td>
+                    <td>
+                        {% if enclosure.isSecurityActive %}
+                            🔒 Security active!
+                        {% else %}
+                            <button class="button button-alarm">Sound alarm !!!</button>
+                        {% endif %}
+                    </td>
                 </tr>
             {% endfor %}
         </table>
diff --git a/web/css/main.css b/web/css/main.css
index c8fe0fb..dde9433 100644
--- a/web/css/main.css
+++ b/web/css/main.css
@@ -7,3 +7,8 @@
     border-bottom: 1px solid #636363;
     text-align: center;
 }
+
+.button-alarm {
+    background: #cb4e56;
+    border-color: #cb4e56;
+}
