diff --git a/app/Resources/views/_partials/_newDinoForm.html.twig b/app/Resources/views/_partials/_newDinoForm.html.twig
new file mode 100644
index 0000000..61b5342
--- /dev/null
+++ b/app/Resources/views/_partials/_newDinoForm.html.twig
@@ -0,0 +1,22 @@
+<form action="{{ url('grow_dinosaur') }}" method="POST">
+    <div class="row">
+        <div class="column">
+            <label for="enclosure">Enclosure</label>
+            <select name="enclosure" id="enclosure">
+                {% for enclosure in enclosures %}
+                    <option value="{{ enclosure.id }}">Enclosure #{{ enclosure.id }}</option>
+                {% endfor %}
+            </select>
+        </div>
+
+        <div class="column">
+            <label for="specification">Dino description</label>
+            <input type="text" id="specification" name="specification" placeholder="Small carnivorous dino friend" />
+        </div>
+
+        <div class="column">
+            <label for="">&nbsp;</label>
+            <input type="submit" class="button" value="Grow dinosaur" />
+        </div>
+    </div>
+</form>
diff --git a/app/Resources/views/default/index.html.twig b/app/Resources/views/default/index.html.twig
index ffc638e..9ebe025 100644
--- a/app/Resources/views/default/index.html.twig
+++ b/app/Resources/views/default/index.html.twig
@@ -1,6 +1,8 @@
 {% extends 'base.html.twig' %}
 
 {% block body %}
+    {{ include('_partials/_newDinoForm.html.twig') }}
+
     <h3>Enclosures</h3>
 
     <table class="table-enclosures">
diff --git a/src/AppBundle/Controller/DefaultController.php b/src/AppBundle/Controller/DefaultController.php
index 944fca4..d05c0c2 100644
--- a/src/AppBundle/Controller/DefaultController.php
+++ b/src/AppBundle/Controller/DefaultController.php
@@ -3,6 +3,8 @@
 namespace AppBundle\Controller;
 
 use AppBundle\Entity\Enclosure;
+use AppBundle\Factory\DinosaurFactory;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\HttpFoundation\Request;
@@ -22,4 +24,32 @@ class DefaultController extends Controller
             'enclosures' => $enclosures,
         ]);
     }
+
+    /**
+     * @Route("/grow", name="grow_dinosaur")
+     * @Method({"POST"})
+     */
+    public function growAction(Request $request, DinosaurFactory $dinosaurFactory)
+    {
+        $manager = $this->getDoctrine()->getManager();
+
+        $enclosure = $manager->getRepository(Enclosure::class)
+            ->find($request->request->get('enclosure'));
+
+        $specification = $request->request->get('specification');
+        $dinosaur = $dinosaurFactory->growFromSpecification($specification);
+
+        $dinosaur->setEnclosure($enclosure);
+        $enclosure->addDinosaur($dinosaur);
+
+        $manager->flush();
+
+        $this->addFlash('success', sprintf(
+            'Grew a %s in enclosure #%d',
+            mb_strtolower($specification),
+            $enclosure->getId()
+        ));
+
+        return $this->redirectToRoute('homepage');
+    }
 }
