diff --git a/tests/AppBundle/Controller/DefaultControllerTest.php b/tests/AppBundle/Controller/DefaultControllerTest.php
index df76e96..978b65b 100644
--- a/tests/AppBundle/Controller/DefaultControllerTest.php
+++ b/tests/AppBundle/Controller/DefaultControllerTest.php
@@ -24,4 +24,21 @@ class DefaultControllerTest extends WebTestCase
         $table = $crawler->filter('.table-enclosures');
         $this->assertCount(3, $table->filter('tbody tr'));
     }
+
+    public function testThatThereIsAnAlarmButtonWithoutSecurity()
+    {
+        $fixtures = $this->loadFixtures([
+            LoadBasicParkData::class,
+            LoadSecurityData::class,
+        ])->getReferenceRepository();
+
+        $client = $this->makeClient();
+
+        $crawler = $client->request('GET', '/');
+
+        $enclosure = $fixtures->getReference('carnivorous-enclosure');
+        $selector = sprintf('#enclosure-%s .button-alarm', $enclosure->getId());
+
+        $this->assertGreaterThan(0, $crawler->filter($selector)->count());
+    }
 }
