diff --git a/app/Resources/views/default/index.html.twig b/app/Resources/views/default/index.html.twig
index 3fabf3b..e61ca97 100644
--- a/app/Resources/views/default/index.html.twig
+++ b/app/Resources/views/default/index.html.twig
@@ -13,6 +13,13 @@
                     <td>
                         Contains <strong>{{ enclosure.dinosaurCount }}</strong> dinosaur(s)
                     </td>
+                    <td>
+                        {% if enclosure.isSecurityActive %}
+                            🔒 Security active!
+                        {% else %}
+                            <button class="button button-alarm">Sound alarm !!!</button>
+                        {% endif %}
+                    </td>
                 </tr>
             {% endfor %}
         </table>
diff --git a/src/AppBundle/Entity/Enclosure.php b/src/AppBundle/Entity/Enclosure.php
index f0d2f67..74af387 100644
--- a/src/AppBundle/Entity/Enclosure.php
+++ b/src/AppBundle/Entity/Enclosure.php
@@ -81,4 +81,15 @@ class Enclosure
     {
         return $this->dinosaurs->count();
     }
+
+    public function getIsSecurityActive(): bool
+    {
+        foreach ($this->securities as $security) {
+            if ($security->getIsActive()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
 }
diff --git a/web/css/main.css b/web/css/main.css
index c8fe0fb..dde9433 100644
--- a/web/css/main.css
+++ b/web/css/main.css
@@ -7,3 +7,8 @@
     border-bottom: 1px solid #636363;
     text-align: center;
 }
+
+.button-alarm {
+    background: #cb4e56;
+    border-color: #cb4e56;
+}
