diff --git a/app/Resources/views/default/index.html.twig b/app/Resources/views/default/index.html.twig
index 4f5a247..ffc638e 100644
--- a/app/Resources/views/default/index.html.twig
+++ b/app/Resources/views/default/index.html.twig
@@ -11,6 +11,13 @@
                     <td>
                         Contains <strong>{{ enclosure.dinosaurCount }}</strong> dinosaur(s)
                     </td>
+                    <td>
+                        {% if enclosure.isSecurityActive %}
+                            🔒 Security active!
+                        {% else %}
+                            <button class="button button-alarm">Sound alarm !!!</button>
+                        {% endif %}
+                    </td>
                 </tr>
             {% endfor %}
         </tbody>
