diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
index 3a8374d..e602dbe 100644
--- a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
@@ -3,12 +3,23 @@
 namespace Tests\AppBundle\Service;
 
 use AppBundle\Entity\Dinosaur;
+use AppBundle\Entity\Enclosure;
 use AppBundle\Entity\Security;
 use AppBundle\Service\EnclosureBuilderService;
+use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 
 class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
 {
+    public function setUp()
+    {
+        $this->truncateEntities([
+            Enclosure::class,
+            Security::class,
+            Dinosaur::class,
+        ]);
+    }
+
     public function testItBuildsEnclosureWithDefaultSpecifications()
     {
         self::bootKernel();
@@ -18,9 +29,7 @@ class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
 
         $enclosureBuilderService->buildEnclosure();
 
-        $em = self::$kernel->getContainer()
-            ->get('doctrine')
-            ->getManager();
+        $em = $this->getEntityManager();
 
         $count = (int)$em
             ->getRepository(Security::class)
@@ -40,4 +49,38 @@ class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
 
         $this->assertSame(3, $count, 'Amount of dinosaurs is not the same');
     }
+
+    private function truncateEntities(array $entities)
+    {
+        $connection = $this->getEntityManager()->getConnection();
+        $databasePlatform = $connection->getDatabasePlatform();
+
+        if ($databasePlatform->supportsForeignKeyConstraints()) {
+            $connection->query('SET FOREIGN_KEY_CHECKS=0');
+        }
+
+        foreach ($entities as $entity) {
+            $query = $databasePlatform->getTruncateTableSQL(
+                $this->getEntityManager()->getClassMetadata($entity)->getTableName()
+            );
+
+            $connection->executeUpdate($query);
+        }
+
+        if ($databasePlatform->supportsForeignKeyConstraints()) {
+            $connection->query('SET FOREIGN_KEY_CHECKS=1');
+        }
+    }
+
+    /**
+     * @return EntityManagerInterface
+     */
+    private function getEntityManager()
+    {
+        self::bootKernel();
+
+        return self::$kernel->getContainer()
+            ->get('doctrine')
+            ->getManager();
+    }
 }
