diff --git a/app/config/config_test.yml b/app/config/config_test.yml
index 2f6d925..7679a47 100644
--- a/app/config/config_test.yml
+++ b/app/config/config_test.yml
@@ -1,6 +1,12 @@
 imports:
     - { resource: config_dev.yml }
 
+services:
+    _defaults:
+        public: true
+
+    test.AppBundle\Service\EnclosureBuilderService: '@AppBundle\Service\EnclosureBuilderService'
+
 framework:
     test: ~
     session:
diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
index fa0e332..2daa544 100644
--- a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
@@ -12,6 +12,6 @@ class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
         self::bootKernel();
 
         $enclosureBuilderService = self::$kernel->getContainer()
-            ->get(EnclosureBuilderService::class);
+            ->get('test.'.EnclosureBuilderService::class);
     }
 }
