diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
new file mode 100644
index 0000000..fa0e332
--- /dev/null
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
@@ -0,0 +1,17 @@
+<?php
+
+namespace Tests\AppBundle\Service;
+
+use AppBundle\Service\EnclosureBuilderService;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
+{
+    public function testItBuildsEnclosureWithDefaultSpecifications()
+    {
+        self::bootKernel();
+
+        $enclosureBuilderService = self::$kernel->getContainer()
+            ->get(EnclosureBuilderService::class);
+    }
+}
