diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
index 2daa544..417ff6a 100644
--- a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
@@ -2,7 +2,10 @@
 
 namespace Tests\AppBundle\Service;
 
+use AppBundle\Entity\Dinosaur;
+use AppBundle\Entity\Security;
 use AppBundle\Service\EnclosureBuilderService;
+use Doctrine\ORM\EntityManager;
 use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 
 class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
@@ -13,5 +16,28 @@ class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
 
         $enclosureBuilderService = self::$kernel->getContainer()
             ->get('test.'.EnclosureBuilderService::class);
+
+        $enclosureBuilderService->buildEnclosure();
+
+        /** @var EntityManager $em */
+        $em = self::$kernel->getContainer()
+            ->get('doctrine')
+            ->getManager();
+
+        $count = (int) $em->getRepository(Security::class)
+            ->createQueryBuilder('s')
+            ->select('COUNT(s.id)')
+            ->getQuery()
+            ->getSingleScalarResult();
+
+        $this->assertSame(1, $count, 'Amount of security systems is not the same');
+
+        $count = (int) $em->getRepository(Dinosaur::class)
+            ->createQueryBuilder('d')
+            ->select('COUNT(d.id)')
+            ->getQuery()
+            ->getSingleScalarResult();
+
+        $this->assertSame(3, $count, 'Amount of dinosaurs is not the same');
     }
 }
