diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
index 3edd1a6..b23289f 100644
--- a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
@@ -5,6 +5,7 @@ namespace Tests\AppBundle\Service;
 use AppBundle\Entity\Dinosaur;
 use AppBundle\Entity\Enclosure;
 use AppBundle\Entity\Security;
+use AppBundle\Factory\DinosaurFactory;
 use AppBundle\Service\EnclosureBuilderService;
 use Doctrine\Common\DataFixtures\Purger\ORMPurger;
 use Doctrine\ORM\EntityManager;
@@ -21,8 +22,18 @@ class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
 
     public function testItBuildsEnclosureWithDefaultSpecifications()
     {
-        $enclosureBuilderService = self::$kernel->getContainer()
-            ->get('test.'.EnclosureBuilderService::class);
+//        $enclosureBuilderService = static::$kernel->getContainer()
+//            ->get('test.'.EnclosureBuilderService::class);
+
+        $dinoFactory = $this->createMock(DinosaurFactory::class);
+        $dinoFactory->expects($this->any())
+            ->method('growFromSpecification')
+            ->willReturn(new Dinosaur());
+
+        $enclosureBuilderService = new EnclosureBuilderService(
+            $this->getEntityManager(),
+            $dinoFactory
+        );
 
         $enclosureBuilderService->buildEnclosure();
 
