diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
index 417ff6a..f504733 100644
--- a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
@@ -3,6 +3,7 @@
 namespace Tests\AppBundle\Service;
 
 use AppBundle\Entity\Dinosaur;
+use AppBundle\Entity\Enclosure;
 use AppBundle\Entity\Security;
 use AppBundle\Service\EnclosureBuilderService;
 use Doctrine\ORM\EntityManager;
@@ -10,19 +11,25 @@ use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 
 class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
 {
-    public function testItBuildsEnclosureWithDefaultSpecifications()
+    public function setUp()
     {
         self::bootKernel();
 
+        $this->truncateEntities([
+            Enclosure::class,
+            Security::class,
+            Dinosaur::class,
+        ]);
+    }
+
+    public function testItBuildsEnclosureWithDefaultSpecifications()
+    {
         $enclosureBuilderService = self::$kernel->getContainer()
             ->get('test.'.EnclosureBuilderService::class);
 
         $enclosureBuilderService->buildEnclosure();
 
-        /** @var EntityManager $em */
-        $em = self::$kernel->getContainer()
-            ->get('doctrine')
-            ->getManager();
+        $em = $this->getEntityManager();
 
         $count = (int) $em->getRepository(Security::class)
             ->createQueryBuilder('s')
@@ -40,4 +47,36 @@ class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
 
         $this->assertSame(3, $count, 'Amount of dinosaurs is not the same');
     }
+
+    private function truncateEntities(array $entities)
+    {
+        $connection = $this->getEntityManager()->getConnection();
+        $databasePlatform = $connection->getDatabasePlatform();
+
+        if ($databasePlatform->supportsForeignKeyConstraints()) {
+            $connection->query('SET FOREIGN_KEY_CHECKS=0');
+        }
+
+        foreach ($entities as $entity) {
+            $query = $databasePlatform->getTruncateTableSQL(
+                $this->getEntityManager()->getClassMetadata($entity)->getTableName()
+            );
+
+            $connection->executeUpdate($query);
+        }
+
+        if ($databasePlatform->supportsForeignKeyConstraints()) {
+            $connection->query('SET FOREIGN_KEY_CHECKS=1');
+        }
+    }
+
+    /**
+     * @return EntityManager
+     */
+    private function getEntityManager()
+    {
+        return self::$kernel->getContainer()
+            ->get('doctrine')
+            ->getManager();
+    }
 }
