diff --git a/app/config/config_test.yml b/app/config/config_test.yml
index 2f6d925..f1d817c 100644
--- a/app/config/config_test.yml
+++ b/app/config/config_test.yml
@@ -14,3 +14,6 @@ web_profiler:
 
 swiftmailer:
     disable_delivery: true
+
+services:
+    test.AppBundle\Factory\DinosaurFactory: '@AppBundle\Factory\DinosaurFactory'
diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
new file mode 100644
index 0000000..2067fd0
--- /dev/null
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceIntegrationTest.php
@@ -0,0 +1,17 @@
+<?php
+
+namespace Tests\AppBundle\Service;
+
+use AppBundle\Service\EnclosureBuilderService;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+class EnclosureBuilderServiceIntegrationTest extends KernelTestCase
+{
+    public function testItBuildsEnclosureWithDefaultSpecifications()
+    {
+        self::bootKernel();
+
+        $enclosureBuilderService = static::$kernel->getContainer()
+            ->get(EnclosureBuilderService::class);
+    }
+}
