diff --git a/tests/AppBundle/Factory/DinosaurFactoryTest.php b/tests/AppBundle/Factory/DinosaurFactoryTest.php
index d04ff56..561c4ad 100644
--- a/tests/AppBundle/Factory/DinosaurFactoryTest.php
+++ b/tests/AppBundle/Factory/DinosaurFactoryTest.php
@@ -13,6 +13,11 @@ class DinosaurFactoryTest extends TestCase
      */
     private $factory;
 
+    /**
+     * @var bool
+     */
+    private $isPopulationControlEnabled = true;
+
     public function setUp()
     {
         $this->factory = new DinosaurFactory();
@@ -32,4 +37,15 @@ class DinosaurFactoryTest extends TestCase
     {
         $this->markTestIncomplete('Waiting for confirmation from GenLab');
     }
+
+    public function testItGrowsASmallVelociraptor()
+    {
+        if ($this->isPopulationControlEnabled) {
+            $this->markTestSkipped();
+        }
+
+        $dinosaur = $this->factory->growVelociraptor(1);
+
+        $this->assertSame(1, $dinosaur->getLength());
+    }
 }
