diff --git a/tests/AppBundle/Entity/DinosaurTest.php b/tests/AppBundle/Entity/DinosaurTest.php
index 5a68ddd..716ba82 100644
--- a/tests/AppBundle/Entity/DinosaurTest.php
+++ b/tests/AppBundle/Entity/DinosaurTest.php
@@ -26,4 +26,14 @@ class DinosaurTest extends TestCase
 
         $this->assertGreaterThan(12, $dinosaur->getLength(), 'Did you put it in the washing machine?');
     }
+
+    public function testReturnsFullSpecificationOfDinosaur()
+    {
+        $dinosaur = new Dinosaur();
+
+        $this->assertSame(
+            'The Unknown non-carnivorous dinosaur is 0 meters long',
+            $dinosaur->getSpecification()
+        );
+    }
 }
