diff --git a/tests/AppBundle/Entity/DinosaurTest.php b/tests/AppBundle/Entity/DinosaurTest.php
index 716ba82..3b3b8d3 100644
--- a/tests/AppBundle/Entity/DinosaurTest.php
+++ b/tests/AppBundle/Entity/DinosaurTest.php
@@ -36,4 +36,16 @@ class DinosaurTest extends TestCase
             $dinosaur->getSpecification()
         );
     }
+
+    public function testReturnsFullSpecificationForTyrannosaurus()
+    {
+        $dinosaur = new Dinosaur('Tyrannosaurus', true);
+
+        $dinosaur->setLength(12);
+
+        $this->assertSame(
+            'The Tyrannosaurus carnivorous dinosaur is 12 meters long',
+            $dinosaur->getSpecification()
+        );
+    }
 }
