diff --git a/src/AppBundle/Entity/Dinosaur.php b/src/AppBundle/Entity/Dinosaur.php
index 8ef630a..67b243e 100644
--- a/src/AppBundle/Entity/Dinosaur.php
+++ b/src/AppBundle/Entity/Dinosaur.php
@@ -15,6 +15,24 @@ class Dinosaur
      */
     private $length = 0;
 
+    /**
+     * @var string
+     * @ORM\Column(type="string")
+     */
+    private $genus;
+
+    /**
+     * @var bool
+     * @ORM\Column(type="boolean")
+     */
+    private $isCarnivorous;
+
+    public function __construct(string $genus = 'Unknown', bool $isCarnivorous = false)
+    {
+        $this->genus = $genus;
+        $this->isCarnivorous = $isCarnivorous;
+    }
+
     public function getLength(): int
     {
         return $this->length;
@@ -27,6 +45,11 @@ class Dinosaur
 
     public function getSpecification(): string
     {
-        return 'The Unknown non-carnivorous dinosaur is 0 meters long';
+        return sprintf(
+            'The %s %scarnivorous dinosaur is %d meters long',
+            $this->genus,
+            $this->isCarnivorous ? '' : 'non-',
+            $this->length
+        );
     }
 }
diff --git a/tests/AppBundle/Entity/DinosaurTest.php b/tests/AppBundle/Entity/DinosaurTest.php
index 15b48a8..3c71e24 100644
--- a/tests/AppBundle/Entity/DinosaurTest.php
+++ b/tests/AppBundle/Entity/DinosaurTest.php
@@ -39,7 +39,7 @@ class DinosaurTest extends TestCase
 
     public function testReturnsFullSpecificationForTyrannosaurus()
     {
-        $dinosaur = new Dinosaur();
+        $dinosaur = new Dinosaur('Tyrannosaurus', true);
 
         $dinosaur->setLength(12);
 
