diff --git a/src/AppBundle/Factory/DinosaurFactory.php b/src/AppBundle/Factory/DinosaurFactory.php
index 2c0ede5..2566479 100644
--- a/src/AppBundle/Factory/DinosaurFactory.php
+++ b/src/AppBundle/Factory/DinosaurFactory.php
@@ -8,10 +8,13 @@ class DinosaurFactory
 {
     public function growVelociraptor(int $length): Dinosaur
     {
-        $dinosaur = new Dinosaur('Velociraptor', true);
+        return $this->createDinosaur('Velociraptor', true, $length);
+    }
 
-        $dinosaur->setLength($length);
+    private function createDinosaur(string $genus, bool $isCarnivorous, int $length)
+    {
+        $dinosaur = new Dinosaur($genus, $isCarnivorous);
 
-        return $dinosaur;
+        $dinosaur->setLength($length);
     }
 }
diff --git a/tests/AppBundle/Factory/DinosaurFactoryTest.php b/tests/AppBundle/Factory/DinosaurFactoryTest.php
index 3f5ef20..4d12e2c 100644
--- a/tests/AppBundle/Factory/DinosaurFactoryTest.php
+++ b/tests/AppBundle/Factory/DinosaurFactoryTest.php
@@ -8,10 +8,19 @@ use PHPUnit\Framework\TestCase;
 
 class DinosaurFactoryTest extends TestCase
 {
+    /**
+     * @var DinosaurFactory
+     */
+    private $factory;
+
+    public function setUp()
+    {
+        $this->factory = new DinosaurFactory();
+    }
+
     public function testItGrowsALargeVelociraptor()
     {
-        $factory = new DinosaurFactory();
-        $dinosaur = $factory->growVelociraptor(5);
+        $dinosaur = $this->factory->growVelociraptor(5);
 
         $this->assertInstanceOf(Dinosaur::class, $dinosaur);
         $this->assertInternalType('string', $dinosaur->getGenus());
