diff --git a/tests/AppBundle/Entity/DinosaurTest.php b/tests/AppBundle/Entity/DinosaurTest.php
index 6af9fc1..5a68ddd 100644
--- a/tests/AppBundle/Entity/DinosaurTest.php
+++ b/tests/AppBundle/Entity/DinosaurTest.php
@@ -17,4 +17,13 @@ class DinosaurTest extends TestCase
 
         $this->assertSame(9, $dinosaur->getLength());
     }
+
+    public function testDinosaurHasNotShrunk()
+    {
+        $dinosaur = new Dinosaur();
+
+        $dinosaur->setLength(15);
+
+        $this->assertGreaterThan(12, $dinosaur->getLength(), 'Did you put it in the washing machine?');
+    }
 }
