diff --git a/src/AppBundle/Entity/Dinosaur.php b/src/AppBundle/Entity/Dinosaur.php
index 3226294..8ef630a 100644
--- a/src/AppBundle/Entity/Dinosaur.php
+++ b/src/AppBundle/Entity/Dinosaur.php
@@ -24,4 +24,9 @@ class Dinosaur
     {
         $this->length = $length;
     }
+
+    public function getSpecification(): string
+    {
+        return 'The Unknown non-carnivorous dinosaur is 0 meters long';
+    }
 }
