diff --git a/tests/AppBundle/Entity/DinosaurTest.php b/tests/AppBundle/Entity/DinosaurTest.php
index 1cf47ae..97c0fbd 100644
--- a/tests/AppBundle/Entity/DinosaurTest.php
+++ b/tests/AppBundle/Entity/DinosaurTest.php
@@ -52,7 +52,9 @@ class DinosaurTest extends TestCase
 
     public function testCanGetAPositionOfDinosaur()
     {
-        $tracker = new Tracker();
+        $tracker = $this->createMock(Tracker::class);
+        $tracker->method('getLocation')
+            ->willReturn(5);
 
         $dinosaur = new Dinosaur();
         $dinosaur->addTracker($tracker);
