diff --git a/src/AppBundle/Entity/Dinosaur.php b/src/AppBundle/Entity/Dinosaur.php
index 165f341..c1af41e 100644
--- a/src/AppBundle/Entity/Dinosaur.php
+++ b/src/AppBundle/Entity/Dinosaur.php
@@ -100,4 +100,9 @@ class Dinosaur
             $this->tracker->getLocation()
         );
     }
+
+    public function incapacitate()
+    {
+        $this->tracker->zap();
+    }
 }
diff --git a/src/AppBundle/Entity/Tracker.php b/src/AppBundle/Entity/Tracker.php
index f6e014c..77ca4c9 100644
--- a/src/AppBundle/Entity/Tracker.php
+++ b/src/AppBundle/Entity/Tracker.php
@@ -20,4 +20,8 @@ class Tracker
         // TODO: replace with real logic before pushing to production :)
         return random_int(0, 1) ? 'yes' : 'no';
     }
+
+    public function zap()
+    {
+    }
 }
diff --git a/tests/AppBundle/Entity/DinosaurTest.php b/tests/AppBundle/Entity/DinosaurTest.php
index 4a7c8ca..91830ea 100644
--- a/tests/AppBundle/Entity/DinosaurTest.php
+++ b/tests/AppBundle/Entity/DinosaurTest.php
@@ -102,4 +102,19 @@ class DinosaurTest extends TestCase
         $this->assertContains('area "3"', $dinosaur->track());
         $this->assertContains('area "3"', $dinosaur->track());
     }
+
+    public function testReceivingExceptionFromAStub()
+    {
+        $tracker = $this->createMock(Tracker::class);
+
+        $tracker->method('zap')
+            ->willThrowException(new \RuntimeException('TODO in the next sprint'));
+
+        $dinosaur = new Dinosaur();
+        $dinosaur->addTracker($tracker);
+
+        $this->expectException(\RuntimeException::class);
+
+        $dinosaur->incapacitate();
+    }
 }
