diff --git a/src/AppBundle/Entity/History.php b/src/AppBundle/Entity/History.php
new file mode 100644
index 0000000..9c70ac3
--- /dev/null
+++ b/src/AppBundle/Entity/History.php
@@ -0,0 +1,42 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="histories")
+ */
+class History
+{
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private $location;
+
+    /**
+     * @var Tracker
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Tracker")
+     */
+    private $tracker;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $createdAt;
+
+    public function __construct(Tracker $tracker)
+    {
+        $this->tracker = $tracker;
+        $this->location = $tracker->getLocation();
+        $this->createdAt = new \DateTime();
+    }
+}
diff --git a/src/AppBundle/Service/TrackerService.php b/src/AppBundle/Service/TrackerService.php
new file mode 100644
index 0000000..6c26165
--- /dev/null
+++ b/src/AppBundle/Service/TrackerService.php
@@ -0,0 +1,33 @@
+<?php
+
+namespace AppBundle\Service;
+
+use AppBundle\Entity\Tracker;
+use Doctrine\ORM\EntityManagerInterface;
+
+class TrackerService
+{
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * @var Tracker[]
+     */
+    private $trackers = [];
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    public function addTracker(Tracker $tracker)
+    {
+        $this->trackers[] = $tracker;
+    }
+
+    public function takeSnapshot()
+    {
+    }
+}
diff --git a/tests/AppBundle/Service/TrackerServiceTest.php b/tests/AppBundle/Service/TrackerServiceTest.php
new file mode 100644
index 0000000..a77faac
--- /dev/null
+++ b/tests/AppBundle/Service/TrackerServiceTest.php
@@ -0,0 +1,31 @@
+<?php
+
+namespace Tests\AppBundle\Service;
+
+use AppBundle\Entity\History;
+use AppBundle\Entity\Tracker;
+use AppBundle\Service\TrackerService;
+use Doctrine\ORM\EntityManagerInterface;
+use PHPUnit\Framework\TestCase;
+
+class TrackerServiceTest extends TestCase
+{
+    public function testItStoresTheLocationOfAllTrackers()
+    {
+        $entityManager = $this->createMock(EntityManagerInterface::class);
+
+        $entityManager->expects($this->exactly(2))
+            ->method('persist')
+            ->with($this->isInstanceOf(History::class));
+
+        $entityManager->expects($this->once())
+            ->method('flush');
+
+        $trackerService = new TrackerService($entityManager);
+
+        $trackerService->addTracker(new Tracker());
+        $trackerService->addTracker(new Tracker());
+
+        $trackerService->takeSnapshot();
+    }
+}
