diff --git a/src/AppBundle/Service/EnclosureBuilderService.php b/src/AppBundle/Service/EnclosureBuilderService.php
index 2bd2f87..cfae4e4 100644
--- a/src/AppBundle/Service/EnclosureBuilderService.php
+++ b/src/AppBundle/Service/EnclosureBuilderService.php
@@ -68,7 +68,7 @@ class EnclosureBuilderService
         for ($i = 0; $i < $numberOfDinosaurs; $i++) {
             $length = $lengths[array_rand($lengths)];
             $specification = "{$length} {$diet} dinosaur";
-            $dinosaur = $this->dinosaurFactory->growFromSpecification([]);
+            $dinosaur = $this->dinosaurFactory->growFromSpecification($specification);
 
             $enclosure->addDinosaur($dinosaur);
         }
