diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceProphecyTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceProphecyTest.php
new file mode 100644
index 0000000..d73d539
--- /dev/null
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceProphecyTest.php
@@ -0,0 +1,42 @@
+<?php
+
+namespace Tests\AppBundle\Service;
+
+use AppBundle\Entity\Dinosaur;
+use AppBundle\Entity\Enclosure;
+use AppBundle\Factory\DinosaurFactory;
+use AppBundle\Service\EnclosureBuilderService;
+use Doctrine\ORM\EntityManagerInterface;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+
+class EnclosureBuilderServiceProphecyTest extends TestCase
+{
+    public function testItBuildsAndPersistsEnclosure()
+    {
+        $em = $this->prophesize(EntityManagerInterface::class);
+
+        $em
+            ->persist(Argument::type(Enclosure::class))
+            ->shouldBeCalledTimes(1);
+
+        $em->flush()->shouldBeCalled();
+
+        $dinoFactory = $this->prophesize(DinosaurFactory::class);
+
+        $dinoFactory
+            ->growFromSpecification(Argument::type('string'))
+            ->shouldBeCalledTimes(2)
+            ->willReturn(new Dinosaur());
+
+        $builder = new EnclosureBuilderService(
+            $em->reveal(),
+            $dinoFactory->reveal()
+        );
+
+        $enclosure = $builder->buildEnclosure(1, 2);
+
+        $this->assertCount(1, $enclosure->getSecurities());
+        $this->assertCount(2, $enclosure->getDinosaurs());
+    }
+}
