diff --git a/src/AppBundle/Factory/DinosaurFactory.php b/src/AppBundle/Factory/DinosaurFactory.php
index 3a29bc8..88a488f 100644
--- a/src/AppBundle/Factory/DinosaurFactory.php
+++ b/src/AppBundle/Factory/DinosaurFactory.php
@@ -23,7 +23,8 @@ class DinosaurFactory
     {
         // defaults
         $codeName = 'InG-' . random_int(1, 99999);
-        $length = $this->lengthDeterminator->getLengthFromSpecification($specification);
+        //$length = $this->lengthDeterminator->getLengthFromSpecification($specification);
+        $length = $this->lengthDeterminator->getLengthFromSpecification('foo');
         $isCarnivorous = false;
 
         if (stripos($specification, 'carnivorous') !== false) {
diff --git a/tests/AppBundle/Factory/DinosaurFactoryTest.php b/tests/AppBundle/Factory/DinosaurFactoryTest.php
index 46b7c98..92e9ccc 100644
--- a/tests/AppBundle/Factory/DinosaurFactoryTest.php
+++ b/tests/AppBundle/Factory/DinosaurFactoryTest.php
@@ -56,7 +56,9 @@ class DinosaurFactoryTest extends TestCase
      */
     public function testItGrowsADinosaurFromSpecification(string $spec, bool $expectedIsCarnivorous)
     {
-        $this->lengthDeterminator->method('getLengthFromSpecification')
+        $this->lengthDeterminator->expects($this->once())
+            ->method('getLengthFromSpecification')
+            ->with($spec)
             ->willReturn(20);
 
         $dinosaur = $this->factory->growFromSpecification($spec);
