diff --git a/src/AppBundle/Service/EnclosureBuilderService.php b/src/AppBundle/Service/EnclosureBuilderService.php
index 88df96d..17b0e23 100644
--- a/src/AppBundle/Service/EnclosureBuilderService.php
+++ b/src/AppBundle/Service/EnclosureBuilderService.php
@@ -57,11 +57,13 @@ class EnclosureBuilderService
     {
         $lengths = ['small', 'large', 'huge'];
         $diets = ['herbivore', 'carnivorous'];
-        $length = $lengths[array_rand($lengths)];
-        $diet = $diets[array_rand($diets)];
-        $specification = "{$length} {$diet} dinosaur";
-        $dinosaur = $this->dinosaurFactory->growFromSpecification($specification);
+        for ($i = 0; $i < $numberOfDinosaurs; $i++) {
+            $length = $lengths[array_rand($lengths)];
+            $diet = $diets[array_rand($diets)];
+            $specification = "{$length} {$diet} dinosaur";
+            $dinosaur = $this->dinosaurFactory->growFromSpecification($specification);
 
-        $enclosure->addDinosaur($dinosaur);
+            $enclosure->addDinosaur($dinosaur);
+        }
     }
 }
diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceTest.php
index ec13f70..54baeb9 100644
--- a/tests/AppBundle/Service/EnclosureBuilderServiceTest.php
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceTest.php
@@ -14,6 +14,10 @@ class EnclosureBuilderServiceTest extends TestCase
         $em = $this->createMock(EntityManagerInterface::class);
         $dinoFactory = $this->createMock(DinosaurFactory::class);
 
+        $dinoFactory->expects($this->exactly(2))
+            ->method('growFromSpecification')
+            ->with($this->isType('string'));
+
         $builder = new EnclosureBuilderService($em, $dinoFactory);
         $enclosure = $builder->buildEnclosure(1, 2);
 
