diff --git a/src/AppBundle/Factory/DinosaurFactory.php b/src/AppBundle/Factory/DinosaurFactory.php
index ceffce5..3a29bc8 100644
--- a/src/AppBundle/Factory/DinosaurFactory.php
+++ b/src/AppBundle/Factory/DinosaurFactory.php
@@ -3,9 +3,17 @@
 namespace AppBundle\Factory;
 
 use AppBundle\Entity\Dinosaur;
+use AppBundle\Service\DinosaurLengthDeterminator;
 
 class DinosaurFactory
 {
+    private $lengthDeterminator;
+
+    public function __construct(DinosaurLengthDeterminator $lengthDeterminator)
+    {
+        $this->lengthDeterminator = $lengthDeterminator;
+    }
+
     public function growVelociraptor(int $length): Dinosaur
     {
         return $this->createDinosaur('Velociraptor', true, $length);
@@ -15,7 +23,7 @@ class DinosaurFactory
     {
         // defaults
         $codeName = 'InG-' . random_int(1, 99999);
-        $length = $this->getLengthFromSpecification($specification);
+        $length = $this->lengthDeterminator->getLengthFromSpecification($specification);
         $isCarnivorous = false;
 
         if (stripos($specification, 'carnivorous') !== false) {
@@ -35,29 +43,4 @@ class DinosaurFactory
 
         return $dinosaur;
     }
-
-    private function getLengthFromSpecification(string $specification): int
-    {
-        $availableLengths = [
-            'huge' => ['min' => Dinosaur::HUGE, 'max' => 100],
-            'omg' => ['min' => Dinosaur::HUGE, 'max' => 100],
-            '😱' => ['min' => Dinosaur::HUGE, 'max' => 100],
-            'large' => ['min' => Dinosaur::LARGE, 'max' => Dinosaur::HUGE - 1],
-        ];
-        $minLength = 1;
-        $maxLength = Dinosaur::LARGE - 1;
-
-        foreach (explode(' ', $specification) as $keyword) {
-            $keyword = strtolower($keyword);
-
-            if (array_key_exists($keyword, $availableLengths)) {
-                $minLength = $availableLengths[$keyword]['min'];
-                $maxLength = $availableLengths[$keyword]['max'];
-
-                break;
-            }
-        }
-
-        return random_int($minLength, $maxLength);
-    }
 }
