diff --git a/src/AppBundle/Service/EnclosureBuilderService.php b/src/AppBundle/Service/EnclosureBuilderService.php
index 2f7fc10..17ac375 100644
--- a/src/AppBundle/Service/EnclosureBuilderService.php
+++ b/src/AppBundle/Service/EnclosureBuilderService.php
@@ -41,6 +41,8 @@ class EnclosureBuilderService
 
         $this->entityManager->persist($enclosure);
 
+        $this->entityManager->flush();
+
         return $enclosure;
     }
 
diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceTest.php
index 7e37c80..0b55dc7 100644
--- a/tests/AppBundle/Service/EnclosureBuilderServiceTest.php
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceTest.php
@@ -19,6 +19,9 @@ class EnclosureBuilderServiceTest extends TestCase
             ->method('persist')
             ->with($this->isInstanceOf(Enclosure::class));
 
+        $em->expects($this->atLeastOnce())
+            ->method('flush');
+
         $dinoFactory = $this->createMock(DinosaurFactory::class);
 
         $dinoFactory->expects($this->exactly(2))
