diff --git a/tests/AppBundle/Service/EnclosureBuilderServiceProphecyTest.php b/tests/AppBundle/Service/EnclosureBuilderServiceProphecyTest.php
index 1df492d..26eda92 100644
--- a/tests/AppBundle/Service/EnclosureBuilderServiceProphecyTest.php
+++ b/tests/AppBundle/Service/EnclosureBuilderServiceProphecyTest.php
@@ -20,5 +20,7 @@ class EnclosureBuilderServiceProphecyTest extends TestCase
 	    ->shouldBeCalledTimes(1);
 
 	$em->flush()->shouldBeCalled();
+
+	$dinoFactory = $this->prophesize(DinosaurFactory::class);
     }
 }
